upstairs/src/volume.rs

```diff
@@ -652,22 +652,32 @@ impl BlockIO for Volume {
         Ok(all_wq)
     }
 
-    // Return a vec of these?
-    // Return a struct with a vec for SV and Some/None for ROP?
-    async fn query_extent_size(&self) -> Result<Block, CrucibleError> {
-        // ZZZ this needs more info, what if ROP and SV differ?
+    // Return a single Vec that contains all the extent_info from the
+    // sub-volumes we have, or if none, then whatever the read only parent
+    // has.  We don't currently support returning extent info if we have both
+    // sub_volumes and a read_only_parent, but as we are the only consumer of
+    // this interface, we can add that if it becomes a requirement.
+    async fn query_extent_info(
+        &self,
+    ) -> Result<Vec<ExtentInfo>, CrucibleError> {
+        let mut extent_info = Vec::new();
         for sub_volume in &self.sub_volumes {
-            match sub_volume.query_extent_size().await {
-                Ok(es) => {
-                    return Ok(es);
+            match sub_volume.query_extent_info().await {
+                Ok(ei) => {
+                    for sv_ei in ei.iter() {
+                        extent_info.push(*sv_ei);
+                    }
```

Comment:
It's a pre-existing condition, but why is it acceptable to skip subvolumes if they return an error here?  It seems like we should return an error.